components/layout/query.rs

```diff
@@ -588,6 +590,13 @@ pub fn process_offset_parent_query(node: ServoLayoutNode<'_>) -> Option<OffsetPa
         .first()
         .cloned()?;
     let mut border_box = fragment.cumulative_box_area_rect(BoxAreaType::Border)?;
+    let cumulative_sticky_offsets = fragment
+        .retrieve_box_fragment()
+        .and_then(|box_fragment| box_fragment.borrow().spatial_tree_node())
+        .map(|node_id| scroll_tree.cumulative_sticky_offsets(node_id));
+    if let Some(offsets) = cumulative_sticky_offsets {
+        border_box = border_box.translate(offsets.map(Au::from_f32_px).cast_unit());
+    }
```

Comment:
This could be a little simpler:

```suggestion
        .map(|node_id| scroll_tree.cumulative_sticky_offsets(node_id))
        .unwrap_or_default();
    border_box = border_box.translate(offsets.map(Au::from_f32_px).cast_unit());
```

Reply:
But `cumulative_sticky_offsets` needs to stay as an Option, because if we can't retrieve the BoxFragment, then the code below shouldn't get the cumulative offsets of the parent.